Core.ApplicationServices/HelpTexts/HelpTextApplicationService.cs

```diff
@@ -0,0 +1,121 @@
+        {
+            _helpTextsRepository = helpTextsRepository;
+            _userContext = userContext;
+            _domainEvents = domainEvents;
+            _helpTextService = helpTextService;
+        }
+
+        public Result<HelpText, OperationError> GetHelpText(string key)
+        {
+            var helpText = this._helpTextsRepository.AsQueryable().FirstOrDefault(ht => ht.Key == key);
+            return helpText != null
+                ? helpText
+                : new OperationError($"Could not find help text with key {key}", OperationFailure.NotFound);
+        }
+
+        public Result<IEnumerable<HelpText>, OperationError> GetHelpTexts()
+        {
+            return Result<IEnumerable<HelpText>, OperationError>.Success(this._helpTextsRepository.AsQueryable().ToList());
+        }
+
+        public Result<HelpText, OperationError> CreateHelpText(HelpTextCreateParameters parameters)
+        {
+            return WithGlobalAdminRights("User is not allowed to create help text")
+                .Match(error => error,
+                    () => WithAvailableKey(parameters.Key)
+                        .Match(error => error,
+                            () =>
+                            {
+                                var newHelpText = new HelpText()
+                                {
+                                    Description = parameters.Description,
+                                    Title = parameters.Title,
+                                    Key = parameters.Key
+                                };
+
+                                _helpTextsRepository.Insert(newHelpText);
+                                _helpTextsRepository.Save();
+                                _domainEvents.Raise(new EntityCreatedEvent<HelpText>(newHelpText));
+
+                                return Result<HelpText, OperationError>.Success(newHelpText);
+                            }));
+        }
+
+        public Maybe<OperationError> DeleteHelpText(string key)
+        {
+            return GetHelpTextWithGlobalAdminRights(key, $"User is not allowed to delete help text with key {key}")
+                        .Match(helpText => {
+                            _helpTextsRepository.Delete(helpText);
+                            _helpTextsRepository.Save();
+                            _domainEvents.Raise(new EntityBeingDeletedEvent<HelpText>(helpText));
+                            return Maybe<OperationError>.None; },
+                            error => error);
+                        
+        }
+
+        public Result<HelpText, OperationError> PatchHelpText(string key, HelpTextUpdateParameters parameters)
+        {
+            return GetHelpTextWithGlobalAdminRights(key, $"User is not allowed to patch help text with key {key}")
+                .Bind(existing => PerformUpdates(existing, parameters)
+                    .Bind(updated =>
```

Comment:
I think you can Bind to the previous bind not inside of it. That way I feel it'll be a bit cleaner

(notice the parentheses)
![image](https://github.com/user-attachments/assets/32dad11a-85b3-4c25-9922-76993ccac86b)